charts/prefect-worker/templates/deployment.yaml

```diff
@@ -93,7 +93,7 @@ spec:
               value: {{ template "worker.apiUrl" . }}
             - name: PREFECT_KUBERNETES_CLUSTER_UID
               value: {{ template "worker.clusterUUID" . }}
-            {{- if eq .Values.worker.apiConfig "cloud" }}
+            {{- if or (eq .Values.worker.apiConfig "cloud") (eq .Values.worker.apiConfig "selfHosted") }}
             - name: PREFECT_API_KEY
               valueFrom:
                 secretKeyRef:
```

Comment:
we'll need to handle this config also! you could create something similar to how we construct the api url in the helpers template for the secret name and key